smolyay/_growth_method.py

```diff
@@ -0,0 +1,98 @@
+import numpy
+
+
+class ClenshawCurtisExponentialGrowthMixin:
```

Comment:
I understand the reasoning to make these mixins, but I am also hesitant to have a mixin modify the state of an object. I usually think of mixins for adding methods but not injecting data.

Since these mixins only contain one function, I recommend to make them a function that returns `num_per_level`. The consumer classes can then call this and do anything else that's needed.

Reply:
The computations to compute end_level and start_level are always identical. I can have this function return all three, or have the nested abstract classes compute those themselves so that computation doesn't have to be reimplemented in every class.